### PR TITLE
2 issues here: 1- if the pager is empty next will launch a StopIterat…

### PR DESCRIPTION
**Pull request by marcbrisson**
2 issues here: 

1- if the pager is empty next will launch a StopIteration exception. So we catch this exception and let it pass. 
2- backoff_ms is actually time in milliseconds, It was given to time.sleep who expects seconds.  So the sleep was way to long. The fix is to convert the milliseconds to seconds

### CODE REVIEW

File: dramatiq_azure/asq.py [hunk 1]
```diff
@@ -130,11 +130,15 @@ def __next__(self) -> Optional[_ASQMessage]:
...
+                    try:
+                        msg_batch = [item for item in next(pager.by_page())]
+                        self.message_cache = [_ASQMessage.from_queue_message(_msg) for _msg in msg_batch]
+                    except StopIteration:
+                        pass
+
                 if not msg_batch:
                     self.misses, backoff_ms = compute_backoff(self.misses, max_backoff=self.timeout)
-                    time.sleep(backoff_ms)
+                    time.sleep(backoff_ms / 1000)
```
Comment (bidossessi) [inline dramatiq_azure/asq.py:141]: compute_backoff already returns the result in milliseconds. See [here](https://github.com/Bogdanp/dramatiq/blob/master/dramatiq/common.py#L47)

**Review by bidossessi — CHANGES_REQUESTED** [review]
See my comments. Thank you for catching the StopIteration. I've been distracted.

**Comment by bidossessi** [comment]
Can you add a test to validate the prefetch behaviour when the queue is empty?

**Comment by marcbrisson** [comment]
Hi Stanislas,

Yes compute_backoff returns in ms, but time.sleep needs seconds  so if we
don't divide by 1000 we are not sending seconds to time.sleep.

100ms /1000 = 0.1s

I see you have merge so I think you are align with this chance

Marc

**Comment by marcbrisson** [comment]
I'll try to do add test for empty queue today

Thanks
Marc
